Modules/Sources/Yosemite/PointOfSale/Items/SearchDebounceStrategy.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+/// Defines the debouncing behavior for search input
+public enum SearchDebounceStrategy: Equatable {
+    /// Smart debouncing: Skip debounce on first keystroke after a search completes, then debounce subsequent keystrokes.
+    /// Optionally delays showing loading indicators until a threshold is exceeded.
+    /// Optimized for slow network searches where the first keystroke should show loading immediately.
+    /// - Parameters:
+    ///   - duration: The debounce duration in nanoseconds for subsequent keystrokes
+    ///   - loadingDelayThreshold: Optional threshold in nanoseconds before showing loading indicators. If nil, shows loading immediately.
+    case smart(duration: UInt64, loadingDelayThreshold: UInt64? = nil)
```

Comment:
Since all cases seem to use the same default, perhaps we can pass it here so we don't need to add it on each instance:
```suggestion
    case smart(duration: UInt64 = 500 * NSEC_PER_MSEC, loadingDelayThreshold: UInt64? = nil)
```